run/pubsub/Run.Samples.Pubsub.MinimalApi/Program.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright 2022 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+var port = Environment.GetEnvironmentVariable("PORT");
+if (port != null)
+{
+    app.Urls.Add($"http://0.0.0.0:{port}");
+}
+
+app.MapPost("/", (Envelope envelope) =>
+{
+    if (envelope?.Message?.Data == null)
+    {
+        app.Logger.LogWarning("Bad Request: Invalid Pub/Sub message format.");
+        return Results.BadRequest();
+    }
+
+    var data = Convert.FromBase64String(envelope.Message.Data);
+    var target = System.Text.Encoding.UTF8.GetString(data);
+
+    app.Logger.LogInformation($"Hello {target}!");
+
+    return Results.NoContent();

```

Comment:
I'm not sure about this, but will this acknowledge the message, does it need to?

Reply:
Yes - the use case in the tutorial uses push notifications and this status code will acknowledge the message.